crates/consensus/auto-seal/src/task.rs

```diff
@@ -200,6 +200,11 @@ where
                                 BlockBody { transactions: body, ommers: vec![], withdrawals: None };
                             header.gas_used = gas_used;
 
+                            // calculate the state root
+                            let state_provider = client.latest().unwrap();
+                            let state_root = state_provider.state_root(post_state.clone()).unwrap();
```

Comment:
you already have the latest state provider initialized for substate
```suggestion
                            let state_root =  substate.db.0.state_root(post_state.clone()).unwrap();
```